core/src/main/java/org/apache/carbondata/core/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,171 @@
+    }
+
+    String type = properties.get(TYPE);
+    if (type != null && !CarbonCommonConstants.GEOHASH.equalsIgnoreCase(type)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be %s for this class.",
+                      CarbonCommonConstants.INDEX_HANDLER, TYPE, CarbonCommonConstants.GEOHASH));
+    }
+
+    properties.put(TYPE, CarbonCommonConstants.GEOHASH);
+
+    String sourceDataTypes = properties.get(SOURCE_COLUMN_TYPES);
+    String[] srcTypes = sourceDataTypes.split(",");
+    for (String srcdataType : srcTypes) {
+      if (!"bigint".equalsIgnoreCase(srcdataType)) {
+        throw new MalformedCarbonCommandException(
+                String.format("%s property is invalid. %s datatypes must be long.",
+                        CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+      }
+    }
+
+    String dataType = properties.get(TARGET_DATA_TYPE);
+    if (dataType != null && !"long".equalsIgnoreCase(dataType)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be long for this class.",
+                      CarbonCommonConstants.INDEX_HANDLER, TARGET_DATA_TYPE));
+    }
+
+    /* Set the generated column data type as long */
+    properties.put(TARGET_DATA_TYPE, "long");
+
+    String originLatitude = properties.get(ORIGIN_LATITUDE);
+    if (originLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s property.",
+                      CarbonCommonConstants.INDEX_HANDLER, ORIGIN_LATITUDE));
+    }
+
+    String minLongitude = properties.get(MIN_LONGITUDE);
+    String maxLongitude = properties.get(MAX_LONGITUDE);
+    String minLatitude = properties.get(MIN_LATITUDE);
+    String maxLatitude = properties.get(MAX_LATITUDE);
+    if (minLongitude == null || maxLongitude == null
+            || minLatitude == null || maxLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s, %s, %s and %s properties.",
+                      CarbonCommonConstants.INDEX_HANDLER, MIN_LONGITUDE, MAX_LONGITUDE,
+                      MIN_LATITUDE, MAX_LATITUDE));
+    }
+
+    String gridSize = properties.get(GRID_SIZE);
+    if (gridSize == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, GRID_SIZE));
+    }
+
+    String conversionRatio = properties.get(CONVERSION_RATIO);
+    if (conversionRatio == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, CONVERSION_RATIO));
+    }
+
+    /* Fill the values to the instance fields */
```

Comment:
need to remove ?

Reply:
Added TODO. This method will be filled by @MarvinLitt in a different PR.